java/src/main/java/com/google/appengine/tools/pipeline/impl/backend/SerializationStrategy.java

```diff
@@ -0,0 +1,34 @@
+package com.google.appengine.tools.pipeline.impl.backend;
+
+import com.google.appengine.tools.pipeline.impl.model.PipelineModelObject;
+
+import java.io.IOException;
+
+public interface SerializationStrategy {
+
+  /**
+   * Given an arbitrary Java Object, returns another object that encodes the
+   * given object but that is guaranteed to be of a type supported by the App
+   * Engine Data Store. Use
+   * {@link #deserializeValue(PipelineModelObject, Object)} to reverse this
+   * operation.
+   *
+   * @param model The model that is associated with the value.
+   * @param value An arbitrary Java object to serialize.
+   * @return The serialized version of the object.
+   * @throws IOException if any problem occurs
+   */
+  Object serializeValue(PipelineModelObject model, Object value) throws IOException;
```

Comment:
I maintained their interface here, but it's weird. the return value is EITHER `Blob`, or a `List<Key>`, each of which refers to a `Blob`.  Possibly we should swap this to something like, `StorableObject` or something, that can maintain strong typing and more clarity here. but open to ideas. 

It's also not *exactly* serialization, as it will actually write values to the datastore in the latter case.